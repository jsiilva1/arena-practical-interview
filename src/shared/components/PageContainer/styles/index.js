import styled from 'styled-components';

import { device } from '../../../../../theme/settings/screens';

export const Container = styled.section`
  width: 90%;
  margin: 20px auto 7% auto;
  display: flex;
  flex-direction: column;
  overflow-wrap: break-word;
  word-wrap: break-word;
  hyphens: auto;
  word-break: break-word;
  
  @media ${device.mobile} {
    width: 100%;
  }

  @media screen
  and (min-width: 420px) 
  and (max-width: 580px) {
    width: 100%;
  }  
`;