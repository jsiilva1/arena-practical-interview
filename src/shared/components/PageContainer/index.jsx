/**
 * External Dependencies
 */
import React from 'react';

/**
 * Internal Dependencies
 */
import { Container } from './styles';

/**
 * Page Container Component
 *
 * @description Render a default container to content of pages
 *
 * @return {*} component
*/
const PageContainer = (props) => {
  const divProps = { ...props };

  // page content
  let content = (
    <Container>
      <div {...divProps} />
    </Container>
  );

  return (
    <>
      <div className="arena-fixed-container">
        {content}
      </div>
    </>
  );
};

export default PageContainer;
