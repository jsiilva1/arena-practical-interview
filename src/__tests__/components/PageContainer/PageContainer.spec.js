import React from 'react';
import { shallow } from 'enzyme';

import PageContainer from '../../../shared/components/PageContainer';

describe('PageContainer Component', () => {
  it('should render a Page Container with content as children correctly', () => {
    const wrapper = shallow(
      <PageContainer>
        <p>Real-time content platform</p>
      </PageContainer>,
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});
