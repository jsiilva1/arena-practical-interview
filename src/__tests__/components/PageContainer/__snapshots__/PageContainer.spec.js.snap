// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer Component should render a Page Container with content as children correctly 1`] = `
<Fragment>
  <div
    className="arena-fixed-container"
  >
    <styled.section>
      <div>
        <p>
          Real-time content platform
        </p>
      </div>
    </styled.section>
  </div>
</Fragment>
`;
